feat: add controlled navigation with randomization

BryanttV opened the pull request:
### Description
This PR adds the functionality to render one by one the child components of a component. Forward and backward buttons are included to navigate between components. Additionally, the following fields were included in the component configuration:
- **Randomness**: Allows the child components to be displayed randomly. This randomization is independent for each user.
- **Next Button Text**: Allows to customize the next button text.
- **Previous Button Text**: Allows to customize previous button text.

### How to Test
1. Install this branch.
2. Add `controlled_navigation` to **Studio** > **Course** > **Settings** > **Advanced Settings**.
3. Add a new component of **Controlled Navigation** in a unit.
4. Add children to the component. As a suggestion, use problem components.

#### Randomness Deactivated (Default)
1. Go to LMS, you can see the first child component, and below it a button to go to the next one.
2. Each component will be rendered in order until the last one is reached.
3. Move backward and forward between components using the **Next** and **Previous** buttons.

#### Randomness Activated
1. Go to LMS, you can see the first child component, and below it a button to go to the next one.
2. Each component will be rendered randomly until the last one is reached. The randomization is unique per user.
3. Move backward and forward between components using the **Next** and **Previous** buttons.

https://github.com/eduNEXT/xblock-controlled-navigation/assets/64033729/2ae4be63-272a-4e99-ac71-c9eb66b782ee

BryanttV committed:
chore: add `edx-opaque-keys` in base requirements

BryanttV committed:
chore: remove unused test

BryanttV committed:
feat: add controlled navigation with randomization

BryanttV committed:
feat: add setting fields to add a custom text in prev and next buttons

BryanttV committed:
chore: move styles to css file

BryanttV committed:
fix: avoid index error when there are no children

BryanttV committed:
chore: include css rule for each navigation button

mariajgrimaldi commented on controlled_navigation/controlled_navigation.py:
```diff
@@ -32,18 +38,70 @@ class XBlockControlledNavigation(
 
+    randomness = Boolean(
+        display_name=_("Randomness"),
+        help=_("When enabled, the children will be displayed in a random order."),
```
```suggestion
        help=_("When enabled, the children of the component will be displayed in a random order."),
```

mariajgrimaldi commented on controlled_navigation/controlled_navigation.py:
```diff
@@ -32,18 +38,70 @@ class XBlockControlledNavigation(
+
+    next_button_text = String(
+        display_name=_("Next Button Text"),
+        help=_("Text for the next button."),
```
```suggestion
        help=_("Text for the next button used to navigate forward through the components' children."),
```
Or something like it?

mariajgrimaldi replied:
Same comment for the previous field.

BryanttV replied:
Done: https://github.com/eduNEXT/xblock-controlled-navigation/commit/3816dcf97e04141542f99d2a3f73b967e7e9ddd5

mariajgrimaldi commented on controlled_navigation/controlled_navigation.py:
```diff
@@ -59,31 +117,170 @@ def author_view(self, context):
+            str: The current child id.
+        """
+        if not self.current_child_id:
+            children = children_ids or self.get_children_ids()
+            self.current_child_id = children[0]
```
does this fail if the children_ids list is empty?

BryanttV replied:
Yes, it would fail. However, the function is never called in case there are no children. Still, I will include a validation

BryanttV replied:
Done: https://github.com/eduNEXT/xblock-controlled-navigation/commit/3816dcf97e04141542f99d2a3f73b967e7e9ddd5

mariajgrimaldi commented on controlled_navigation/controlled_navigation.py:
```diff
@@ -59,31 +117,170 @@ def author_view(self, context):
+        """
+        children = self.get_children_ids()
+        current_index = children.index(self.get_current_child_id())
+        if current_index == 0:
+            return {"result": "error"}
```
[curious] Why is this handled as an error?

BryanttV replied:
It indicates that you cannot go back because there are no more children. The same for when you want to move forward.

mariajgrimaldi reviewed (commented):
A few comments for you to address!

mariajgrimaldi reviewed (approved):
I tested it locally and it worked as expected! Thank you.
 Just a few comments left to address:
- The next/previous button need more padding
- Add more explanation text to the help field attribute

BryanttV committed:
chore: address PR review

BryanttV committed:
chore: add margin in button container

BryanttV commented:
Thanks! @mariajgrimaldi

- I updated the texts here: https://github.com/eduNEXT/xblock-controlled-navigation/commit/3816dcf97e04141542f99d2a3f73b967e7e9ddd5
- I added margin in the buttons container: https://github.com/eduNEXT/xblock-controlled-navigation/commit/44050a5552385e3e06924f948b02c60b185ebf9c
